### PR TITLE
[12.0][ADD] purchase_manual_delivery

### PR DESCRIPTION
**Pull request by AdriaGForgeFlow**
This PR adds the module purchase_manual_delivery in v12.

The goal of this module is to allow the manual creation of incoming shipments. When installed,
a purchase order won't impact directly the stock. It will not create the pickings upon order confirmation.
It allows the reception and the impact on stock to be done manually when needed.
The goal is to be used when a supplier does not confirm the full order at once. As the supplier
confirms the different purchase order lines, the user can manually create the incoming
shipments for those specific lines.

cc ~ @forgeflow

### CODE REVIEW

File: purchase_manual_delivery/models/purchase_order.py [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+    _inherit = "purchase.order.line"
+
+    existing_qty = fields.Float(
+        compute="_compute_existing_qty",
+        string="Existing Qty",
+        digits=dp.get_precision("Product Unit of Measure"),
+        help="Quantity already planned or shipped (stock movements "
+        "already created)",
+    )
+    pending_qty_to_receive = fields.Boolean(
```
Comment (HviorForgeFlow) [inline purchase_manual_delivery/models/purchase_order.py]: I would use:
```suggestion
    pending_to_receive = fields.Boolean(
```
I doesn't not make sense to me to use quantity on a boolean, could be confusing.

Comment (AdriaGForgeFlow) [reply purchase_manual_delivery/models/purchase_order.py]: Done :+1:

**Review by AaronHForgeFlow — APPROVED** [review]
:+1: Functional tests. Works like a charm. And the feature to select existing pickings is awesome :medal_sports:

File: purchase_manual_delivery/__manifest__.py [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+{
+    'name': 'Purchase Manual Delivery',
+    'summary': """
+        Prevents pickings to be auto generated upon Purchase Order confirmation
+        and adds the ability to manually generate them as the supplier confirms
+        the different purchase order lines.
+    """,
+    'version': '12.0.1.0.0',
+    'license': 'AGPL-3',
+    'author': 'Eficent Business and IT Consulting Services S.L.,'
```
Comment (AaronHForgeFlow) [inline purchase_manual_delivery/__manifest__.py]: Please, ForgeFlow everywhere :grin:

Comment (AdriaGForgeFlow) [reply purchase_manual_delivery/__manifest__.py]: Thanks for the heads up :+1:

**Review by HviorForgeFlow — APPROVED** [review]
Functional review
:+1:

**Comment by HviorForgeFlow** [comment]
@AdriaGForgeFlow squash commits

**Comment by AdriaGForgeFlow** [comment]
> @AdriaGForgeFlow squash commits

Done :+1:

**Review by HviorForgeFlow — APPROVED** [review]

**Comment by rvalyi** [comment]
A nice use case I think is when you have electronic invoicing and you may want to create the incoming shipments based on the supplier electronic invoice. I'll review it, thanks!

**Pull request merged**